nilearn/plotting/html_connectome.py

```diff
@@ -27,8 +27,33 @@ def _prepare_line(edges, nodes):
     return path_edges, path_nodes
 
 
+def _prepare_colors(marker_color, path_nodes):
+    marker_colors, marker_colorscale = None, None
+
+    if isinstance(marker_color, str) and marker_color == 'auto':
+        marker_colors = np.linspace(0, 100, len(path_nodes)).astype("uint8")
+        marker_colors = marker_colors[path_nodes].tolist()
+        marker_colorscale = 'Viridis'
+    elif isinstance(marker_color, str):
+        color_as_hex = mpl_colors.to_hex(marker_color)
+        marker_colorscale = [[0, color_as_hex], [1, color_as_hex]]
+        marker_colors = [0] * len(path_nodes)
+    elif isinstance(marker_color, list):
+        if len(marker_color) != np.max(path_nodes) + 1:
+            raise ValueError("Number of colors provided should match the number of nodes")
```

Comment:
this will be raised if some nodes are disconnected from all the others.

I think that would be another motivation for separating the line and markers

Reply:
my suggestion is to treat the line and the markers separately. ie instead of
having only "con_x", "con_y", "con_z", have "con_x", "con_y", "con_z",
"marker_x", "marker_y", "marker_z" (or any better names you can think of -- we
could rename "con_x" -> "line_x" for example)

they won't have the same size because the line goes through all edges, whereas
the list of markers goes through each node exactly once.

note storing the connectome nodes twice is not a problem, what thakes space is
the brain mesh anyway, not the connectome.

then in connectome_plot_template.html, there wouldn't be 2 distinct functions
`addMarkers` and `addConnectome` any more -- there would be only one call to
`Plotly.plot`. but the array of traces would contain 2 traces.
instead of `Plotly.plot("connectome-plot", [{mode: "lines+markers+text", ...}])`
we would have
`Plotly.plot("connectome-plot", [{mode: "lines", ...}, {mode: "markers+text, ...}])`

it will solve the question of showing markers that don't have any connection,
avoid plotting several times the same marker on top of itself, and make it
simpler to pass the colors and sizes for the markers as these lists will now
have the same length as `node_coords`.

does that make sense?